src/encoded/types/experiment.py

```diff
@@ -382,29 +411,50 @@ def superseded_by(self, request, superseded_by):
+        # That replicate should have a libraries property which, as calculated
+        # in replicate.libraries (ENCD-4251), should have collected all
+        # possible technique replicates belong to the biological replicate.
+        # TODO: change this once we remove technical_replicate_number.
+        bio_rep_dict = {}
+        for rep in replicates:
+            replicateObject = request.embed(rep, '@@object')
+            if replicateObject['status'] == 'deleted':
+                continue
+            bio_rep_num = replicateObject['biological_replicate_number']
+            if bio_rep_num not in bio_rep_dict:
+                bio_rep_dict[bio_rep_num] = replicateObject
+                continue
+            tech_rep_num = replicateObject['technical_replicate_number']
+            if tech_rep_num < bio_rep_dict[bio_rep_num]['technical_replicate_number']:
+                bio_rep_dict[bio_rep_num] = replicateObject
+
         # Compare the biosamples to see if for humans they are the same donor and for
         # model organisms if they are sex-matched and age-matched
-        biosample_dict = {}
         biosample_donor_list = []
         biosample_number_list = []
 
-        for rep in replicates:
-            replicateObject = request.embed(rep, '@@object')
-            if replicateObject['status'] == 'deleted':
-                continue
-            if 'library' in replicateObject:
-                libraryObject = request.embed(replicateObject['library'], '@@object')
-                if 'biosample' in libraryObject:
-                    biosampleObject = request.embed(libraryObject['biosample'], '@@object')
-                    biosample_dict[biosampleObject['accession']] = biosampleObject
-                    biosample_donor_list.append(biosampleObject.get('donor'))
-                    biosample_number_list.append(replicateObject.get('biological_replicate_number'))
-                    biosample_species = biosampleObject.get('organism')
-                    biosampleTypeObject = request.embed(
-                        biosampleObject['biosample_ontology'],
-                        '@@object'
-                    )
-                    biosample_type = biosampleTypeObject.get('classification')
+        for replicateObject in bio_rep_dict.values():
+            if 'libraries' in replicateObject and replicateObject['libraries']:
+                biosamples = request.select_distinct_values(
+                    'biosample', *replicateObject['libraries']
+                )
+                if biosamples:
+                    for b in biosamples:
+                        biosampleObject = request.embed(b, '@@object')
+                        biosample_donor_list.append(
+                            biosampleObject.get('donor')
+                        )
+                        biosample_number_list.append(
+                            replicateObject.get('biological_replicate_number')
```

Comment:
What happens if this doesn't exist for some reason and you append None?

Reply:
"biological_replicate_number" is required for replicate objects here.
Biosample object not having donor should be handled properly by the old code below, right?
https://github.com/ENCODE-DCC/encoded/blob/ENCD-4251-tech-rep-yl/src/encoded/types/experiment.py#L478-L495